polarrouteserver/frontend/callbacks.py

```diff
@@ -0,0 +1,457 @@
+        Input("marker-store", "modified_timestamp"),
+        State("marker-store", "data"),
+    )
+    def update_coords(ts, marker_data):
+        """Updates the coordinates text boxes after a change to the marker store."""
+
+        if ts is None or marker_data == {}:
+            return no_update, no_update
+
+        markers = []
+        for loc in marker_data.keys():
+            markers.append(
+                marker(marker_data[loc]["lat"], marker_data[loc]["lon"], loc)
+            )
+
+        if marker_data.get("start", None) is None:
+            start_value = no_update
+        else:
+            start_value = (
+                f"{marker_data['start']['lat']:.2f}, {marker_data['start']['lon']:.2f}"
+            )
+
+        if marker_data.get("end", None) is None:
+            end_value = no_update
+        else:
+            end_value = (
+                f"{marker_data['end']['lat']:.2f}, {marker_data['end']['lon']:.2f}"
+            )
+
+        return start_value, end_value
+
+    @app.callback(
+        Output("mouse-coords-container", "children"),
+        Input("map", "mouseCoords"),
+        prevent_initial_call=True,
+    )
+    def update_mouse_coords(coords):
+        lat = coords["area"]["lat"]
+        lon = coords["area"]["lng"]
+
+        return f"({lat:.2f}, {lon:.2f})"
+
+    @app.callback(
+        Output("amsr-overlay", "children"),
+        Input("amsr-date-slider", "value"),
+        prevent_initial_call=True,
+    )
+    def update_amsr_overlay(slider_value):
+        sic_date = default_sic_date - datetime.timedelta(days=abs(slider_value))
+        return amsr_layer(sic_date)
+
+    @app.callback(
+        Output("routes-fg", "children"),
+        Input("route-visibility-store", "modified_timestamp"),
+        State("route-visibility-store", "data"),
+        State("routes-store", "data"),
+    )
+    def update_map_routes(ts, route_visibility, routes_data):
+        features = []
+
+        if ts is None:
+            return no_update
+
+        for r in route_visibility:
+            route = [x for x in routes_data if x["id"] == r["id"]][0]
+
+            if r.get("fuel") or r.get("traveltime"):
+                features.append(
+                    marker(
+                        lat=route["start_lat"],
+                        lon=route["start_lon"],
+                        loc="start",
+                        draggable=False,
+                        interactive=False,
+                        opacity=0.6,
+                    )
+                )
+                features.append(
+                    marker(
+                        lat=route["end_lat"],
+                        lon=route["end_lon"],
+                        loc="end",
+                        draggable=False,
+                        interactive=False,
+                        opacity=0.6,
+                    )
+                )
+
+            if r.get("fuel") and len(route.get("json")) > 0:
```

Comment:
The hover-over tooltip shows the travel-optimised information for the fuel optimised route. Just the indexing flipped I think
```suggestion
            if r.get("fuel") and len(route.get("json")) > 1:
```